pkg/controllers/dynakube/injection/reconciler.go

```diff
@@ -120,6 +125,17 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 	return nil
 }
 
+func (r *reconciler) reconcileIstioForCSIDriver(ctx context.Context) error {
+	if r.istioReconciler != nil {
+		err := r.istioReconciler.ReconcileCSIDriver(ctx, r.dynakube)
+		if err != nil {
+			return errors.WithMessage(err, "failed to reconcile istio objects for CSI Driver")
+		}
+	}
+
+	return nil
+}
```

Comment:
we already have a `r.istioReconciler != nil` check, why have it separately?

Reply:
The only check I see is the one for `ReconcileCodeModuleCommunicationHosts` which comes after the `connectionInfoReconciler`. The CSIDriver reconciler comes before it, how would you like to have this?

Reply:
why does the CSI reconcile need to before the `connectionInfoReconciler`? nothing depends on the CSI reconcile (other then the csi-driver, but that is in a whole different pod, so I wouldn't worry about it)
- The csi-driver also depends on the `r.pmcSecretreconciler.Reconcile(ctx)` which is at the very end of the injection reconcile, so 😛 

the only "requirement" is that the CSI reconcile happens after the version reconcile

Reply:
You are totally right, thanks

https://github.com/Dynatrace/dynatrace-operator/pull/3343/commits/eeb60eb316cf1d7799d179f74d01bbe780ee3cef